pkg/credentials/filecache.go

```diff
@@ -36,6 +38,16 @@ func (r *RealClock) Now() time.Time {
 	return time.Now()
 }
 
+// Flock provides an interface to handle file locking.
+//counterfeiter:generate -o fakes/fake_flock.go . Flock
+type Flock interface {
+	TryRLockContext(ctx context.Context, retryDelay time.Duration) (bool, error)
+	TryLockContext(ctx context.Context, retryDelay time.Duration) (bool, error)
```

Comment:
whats the difference?

Reply:
I specifically didn't want to mock flock because I wanted to make sure that the ability is working with a proper file system. That's why the tests contained a temp folder in which things were happening.

Reply:
Also if you mock the locking mechanism and the file system means that you aren't testing a large junk of the behaviour.

Reply:
>whats the difference?

`TryRLockContext` takes a shared lock, whereas `TryLockContext` takes an exclusive lock.

Reply:
@cPu1 i missed this earlier but just leaving a comment here as a reminder, I think we should add some comments here explaining what these functions do. They look identical and are likely to cause confusion.

Reply:
I have added godoc comments.